src/java/htsjdk/samtools/sra/SRAAccession.java

```diff
@@ -0,0 +1,80 @@
+/*===========================================================================
+*
+*                            PUBLIC DOMAIN NOTICE
+*               National Center for Biotechnology Information
+*
+*  This software/database is a "United States Government Work" under the
+*  terms of the United States Copyright Act.  It was written as part of
+*  the author's official duties as a United States Government employee and
+*  thus cannot be copyrighted.  This software/database is freely available
+*  to the public for use. The National Library of Medicine and the U.S.
+*  Government have not placed any restriction on its use or reproduction.
+*
+*  Although all reasonable efforts have been taken to ensure the accuracy
+*  and reliability of the software and data, the NLM and the U.S.
+*  Government do not and cannot warrant the performance or results that
+*  may be obtained by using this software or data. The NLM and the U.S.
+*  Government disclaim all warranties, express or implied, including
+*  warranties of performance, merchantability or fitness for any particular
+*  purpose.
+*
+*  Please cite the author in any work or product based on this material.
+*
+* ===========================================================================
+*
+*/
+
+package htsjdk.samtools.sra;
+
+import htsjdk.samtools.util.Log;
+import gov.nih.nlm.ncbi.ngs.NGS;
+
+public class SRAAccession {
+    private static final Log log = Log.getInstance(SRAAccession.class);
+
+    private static Boolean isSupportedCached = null;
+    private static String appVersionString = null;
+    private final static String htsJdkVersionString = "HTSJDK-NGS";
+    private final static String defaultAppVersionString = "[unknown software]";
+
+    private String acc;
+
+    public static void setAppVersionString(String appVersionString) {
+        SRAAccession.appVersionString = appVersionString;
+    }
+
+    public static boolean isSupported() {
+        if (isSupportedCached == null) {
+            log.debug("Checking if SRA module is supported in that environment");
+            isSupportedCached = NGS.isSupported();
+            if (!isSupportedCached) {
+                log.info("SRA is not supported. Will not able to read from SRA");
+            } else {
+                String versionString = appVersionString == null ? defaultAppVersionString : appVersionString;
+                versionString += " through " + htsJdkVersionString;
+                NGS.setAppVersionString(versionString);
```

Comment:
This should happen very early during application initialization. Once isSupportedCached is set, it is never be modified and version string is not touched.

So, that should be thread-safe assuming early call to isSupported or to isValid (which calls isSupported internally) during application initialization